tests/unit/testsuite/Bolt/Boltpay/Block/Checkout/BoltpayTest.php

```diff
@@ -1,1056 +1,2332 @@
      */
-    public function testBuildCartDataWithEmptyTokenField()
+    public function getTrackJsUrl_inProduction_returnsJSUrlFromHelperWithSuffix()
     {
-        $autoCapture = true;
-        $this->app->getStore()->setConfig('payment/boltpay/auto_capture', $autoCapture);
-
-        // Prepare test response object
-        $testBoltResponse = new stdClass();
-
-        $result = $this->currentMock->buildCartData($testBoltResponse);
-
-        $cartData = array(
-            'orderToken'    => '',
+        TestHelper::stubConfigValue('payment/boltpay/test', 0);
+        $this->assertEquals(
+            'https://connect.bolt.com/track.js',
+            $this->currentMock->getTrackJsUrl()
         );
-
-        $this->assertEquals($cartData, $result, 'Something wrong with testBuildCartDataWithEmptyTokenField()');
     }
 
     /**
-     * @inheritdoc
-     * @group Block
+     * Setup method for tests covering {@see Bolt_Boltpay_Block_Checkout_Boltpay::getCartDataJs}
+     *
+     * @param string $checkoutType
+     * @return array consisting of quote and current class mock
+     * @throws Mage_Core_Exception if unable to stub Boltpay helper
+     * @throws Exception if test class name is not defined
      */
-    public function testBuildCartDataWithoutAutoCapture()
+    private function getCartDataJsSetUp($checkoutType)
     {
-        $autoCapture = false;
-        $this->app->getStore()->setConfig('payment/boltpay/auto_capture', $autoCapture);
-
-        // Prepare test response object
-        $testBoltResponse = new stdClass();
-        $testBoltResponse->token = md5('bolt');
-
-        $result = $this->currentMock->buildCartData($testBoltResponse);
+        $quote = Mage::getModel('sales/quote');
+        TestHelper::stubHelper('boltpay', $this->boltHelperMock);
+        $currentMock = $this->getTestClassPrototype(true)
+            ->setMethods(
+                array(
+                    'getSessionQuote',
+                    'getAddressHints',
+                    'getReservedUserId',
+                    'isEnableMerchantScopedAccount',
+                    'buildBoltCheckoutJavascript'
+                )
+            )
+            ->getMock();
+        $currentMock->expects($this->once())->method('getSessionQuote')->with($checkoutType)->willReturn($quote);
+        return array($quote, $currentMock);
+    }
 
-        $cartData = array(
-            'orderToken' => md5('bolt'),
+    /**
+     * @test
+     * that getCartDataJs adds signed_merchant_user_id to hint data if reserved user id exists in session
+     * and merchant scoped account is enabled and proceeds to build Bolt checkout javascript
+     *
+     * @covers ::getCartDataJs
+     * @covers ::getAddressHints
+     *
+     * @throws Mage_Core_Exception if unable to setup test dependencies
+     */
+    public function getCartDataJs_withReservedCustomerIdAndMerchantScopedAccountEnabled_buildsBoltCheckoutJSWithSignedMerchantUserId()
```

Comment:
Yikes this is a ridiculously long function name. Is there really no way to make this shorter? I would much prefer a shorter name with comments (which you already have) than an overly verbose one.

Reply:
In general the function names are a bit too long imo. I think after a certain point it's better to have shorter names, even if they don't describe 100% what's going on, and have comments for clarification.

Reply:
hmm... let's see if I can get creative here.  The title is accurate. :)

Reply:
yes, we knew that this would happen when we were discussing the spec.  My preference was for comments, and Alex's was for long descriptive names.  I will try to get creative.

Reply:
See, I don't mind long descriptive names, but only up to a certain length limit. For instance "getSessionObject_withAdminCheckoutType_returnsAdminSessionQuote" is acceptable to me. Imo it's ok to not have the name be always 100% accurate as long as there are comments that make it clear what the method is doing. After all that's what comments are for. I'll try to get Alex to change his mind tomorrow. Maybe he will after seeing this monster of a name lol.

Reply:
Though I distinctively remember him saying in the meeting we had to discuss naming conventions that if the name gets too long we can cut it short and add comments. I'll check with him tomorrow. If that's the case the Notion doc will need to be updated.

Reply:
I think we are or will all be in agreement based on this PR and this discussion:
![image](https://user-images.githubusercontent.com/973856/72865013-d4711680-3c9b-11ea-9c0b-971a909c288f.png)

![image](https://user-images.githubusercontent.com/973856/72865063-0f734a00-3c9c-11ea-81eb-b627e0acd61d.png)

![image](https://user-images.githubusercontent.com/973856/72865331-0a62ca80-3c9d-11ea-856e-46b62bb69a11.png)

Reply:
However, I think that this will resolve the issue: